src/Confluent.Kafka/Impl/NativeMethods/NativeMethods_Debian9.cs

```diff
@@ -0,0 +1,360 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Text;
+using System.Runtime.InteropServices;
+using Confluent.Kafka.Internal;
+
+
+namespace Confluent.Kafka.Impl.NativeMethods
+{
+    /// <summary>
+    ///     This class should be an exact replica of other NativeMethods classes, except
+    ///     for the DllName const.
+    /// </summary>
+    /// <remarks>
+    ///     This copy/pasting is required because DllName must be const. 
+    ///     TODO: generate the NativeMethods classes at runtime (compile C# code) rather
+    ///     than copy/paste.
+    /// 
+    ///     Alternatively, we could have used dlopen to load the native library, but to 
```

Comment:
Is this statement still correct? I thought we were infact using dlopen()

Reply:
this comment is still close enough to correct I think. there is now the option of the user specifying a path, in which case `dlopen` is used. If not, the attribute method (that this comment is referring to) is used.